### ISSUE TITLE
Fish is not detected and sometimes causes a memory leak

### ISSUE THREAD

**Issue by feskyde** [dylanaraps/neofetch#531]
## Description

Title explains all, I guess.

#### Neofetch version

``` bash
casa@solarium ~> neofetch --version 
Neofetch 2.0
```

## Screenshot

Doesn't show anything after this:

![screenshot from 2016-12-14 15-04-23](https://cloud.githubusercontent.com/assets/20542037/21194425/bc4cf160-c20e-11e6-9ec5-f21231fdca1b.png)

## Verbose log

After being 5 minutes idle:

http://hastebin.com/divecoxequ

**Comment by dylanaraps** [comment]
I can't reproduce this.

- How often does this occur?
- Does this command consistently work? This is the command we run to get shell version. Shell name is just `$SHELL`. This is the only command we run in the shell function (for fish) and the only reason it could hang. 
    - `fish -c 'printf "%s" "$FISH_VERSION"'`
- What's the output of `echo "$FISH_VERSION"`

**Comment by feskyde** [comment]
- It only happens when using Fish as default terminal:

![](http://i.imgur.com/ZfqJpYp.png)

- `fish -c 'printf "%s" "$FISH_VERSION"'` and `echo "$FISH_VERSION"` returns `2.4.0`

**Comment by dylanaraps** [comment]
Does running `fish -c 'printf "%s" "$FISH_VERSION"'` when fish is your default shell make your terminal hang?

**Comment by feskyde** [comment]
Doesn't if I remove `neofetch` from the Fish configuration

**Comment by dylanaraps** [comment]
How have you configured `fish` to use `neofetch`? 

This doesn't seem like a neofetch issue to me.

**Comment by feskyde** [comment]
> How have you configured fish to use neofetch?

I added `neofetch --disable packages` to the end of `~/.config/fish/config.fish`

> This doesn't seem like a neofetch issue to me.

I will raise an issue to Fish developers, just in case 😄

**Comment by dylanaraps** [comment]
I think I've figured out the cause of this.

This is happening because Neofetch is in your `config.fish`.

1. You launch Neofetch
2. Neofetch calls `fish -c 'printf "%s" "$FISH_VERSION"'` to get the shell version.
3. This spawns **another** `fish` shell and since Neofetch is in your `config.fish` this spawns **another** Neofetch.
4. This second Neofetch process calls `fish -c 'printf "%s" "$FISH_VERSION"'` which spawns a **third** Neofetch and the cycle continues.
5. Infinite Fish and Neofetch processes are started causing the memory leak.

I'm working on a fix now.

**Comment by dylanaraps** [comment]
Fixed. 

What a weird bug. :P

**Comment by Litarvan** [comment]
Same here on Neofetch 2.0.2 :/

**Comment by dylanaraps** [comment]
Yup, this was fixed after the release of 2.0.2. The fix will be available in 2.1.0.

**Comment by Litarvan** [comment]
Here how i managed to make it work :)

(config.fish)
```fish
if not set -q $fetchDone
    neofetch
    set $fetchDone true
end
```
